packages/clay-navigation-bar/src/index.tsx

```diff
@@ -0,0 +1,161 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React from 'react';
+
+import ClayIcon from '@clayui/icon';
+import ClayLink from '@clayui/link';
+import warning from 'warning';
+
+import {useTransition} from './hooks';
+
+interface ItemProps extends React.HTMLAttributes<HTMLLIElement> {
+	/**
+	 * Determines the active state of an dropdown list item.
+	 */
+	active?: boolean;
+
+	/**
+	 * Children elements received from ClayNavigationBar.Item component.
+	 */
+	children: React.ReactElement;
+}
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Children elements received from ClayNavigationBar component.
+	 */
+	children: React.ReactElement<ItemProps>[];
+
+	/**
+	 * Determines the style of the Navigation Bar
+	 */
+	inverted?: boolean;
+
+	/**
+	 * Path to the location of the spritemap resource.
+	 */
+	spritemap: string;
+
+	/**
+	 * Set up dropdown's trigger label.
+	 */
+	triggerLabel: string;
+}
+
+type ItemType = React.FunctionComponent<ItemProps>;
+
+const Item: ItemType = ({active, children, className, ...otherProps}) => {
+	return (
+		<li
+			{...otherProps}
+			className={classNames('nav-item', {
+				active,
+				className,
+			})}
+		>
+			{React.Children.map(
+				children,
+				(child: React.ReactElement<ItemProps>, index) =>
+					React.cloneElement(child, {
+						...child.props,
```

Comment:
Since we are accepting anything within our Item, as a simple button, I am wondering if this can have problems with this and force using `child.props.className` too, you can check out these use cases.

Reply:
Makes sense Matu! I'm thinking about passing `active` via context instead the className.

Reply:
I added some examples using HTMLElements as child of `<ClayNavigationBar.Item>` and everything worked fine.